main: rename `list-images` to `list` (with alias for the old names)

supakeen opened the pull request:
Rename the `list-images` command to `list`. We don't have `-image(s)` in our other subcommands so this is for consistency.

We keep a `list-images` alias behind for compatibility reasons.

---

Closes #78, name can be discussed but the initial issue didn't garner attention.

bcl reviewed (dismissed):
Looks good to me

mvo5 commented:
I'm not 100% on-board (but also not 100% opposed) (but maybe just because of habbit by now).

The argument against would be that the (mandatory) argument of build gives the information away `image-builder build <image-type>` but for `list` there is no argument. But if we do it we should also do it for describe-image -> describe (just as outlined in the issue).

supakeen commented:
> I'm not 100% on-board (but also not 100% opposed) (but maybe just because of habbit by now).
> 
> The argument against would be that the (mandatory) argument of build gives the information away `image-builder build <image-type>` but for `list` there is no argument. But if we do it we should also do it for describe-image -> describe (just as outlined in the issue).

Under that reasoning `describe` follows the same pattern as `build` (it takes an image as its direct argument) so that one would be even more consistent! I've added a commit that also renames `describe` (and leaves `describe-image` as its alias).

supakeen commented:
FWIW, I'm fine to drop the `list-images` commit here if that's the controversial bit though it would be the "odd one out". My feeling is that *all* these subcommands already implicitly interact on images, mostly because I can't really think of a different meaning for `image-builder list` than "lists images".

mvo5 commented:
> FWIW, I'm fine to drop the `list-images` commit here if that's the controversial bit though it would be the "odd one out". My feeling is that _all_ these subcommands already implicitly interact on images, mostly because I can't really think of a different meaning for `image-builder list` than "lists images".

I mean, we could simply have two PRs, merge the uncontroversial one, then ponder a bit about the other one (and hear more feedback). 

(unrelated - I'm also contemplating making `describe` no longer hidden, with the `partition_table` addition and the (soon hopefully) converage of the output with our image type yaml we hopefully it gets more and more non-specialized I feel)

supakeen commented:
> > FWIW, I'm fine to drop the `list-images` commit here if that's the controversial bit though it would be the "odd one out". My feeling is that _all_ these subcommands already implicitly interact on images, mostly because I can't really think of a different meaning for `image-builder list` than "lists images".
> 
> I mean, we could simply have two PRs, merge the uncontroversial one, then ponder a bit about the other one (and hear more feedback).
> 
> (unrelated - I'm also contemplating making `describe` no longer hidden, with the `partition_table` addition and the (soon hopefully) converage of the output with our image type yaml we hopefully it gets more and more non-specialized I feel)

Split into #186 and dropped the commit from here.

bcl reviewed (approved):
Still ok with me :)

mvo5 commented:
Thanks, I'm fine with the rename then (still not 100% but the old name is also not 100%)  there is probably nothing 100% fitting and both apt and dnf have list to show their packages so this seems analogous (enough).
